gateway/src/apicast/policy/3scale_referrer/3scale_referrer.lua

```diff
@@ -0,0 +1,20 @@
+local policy = require('apicast.policy')
+local _M = policy.new('3scale Referrer policy')
+
+local new = _M.new
+
+function _M.new(config)
+  local self = new(config)
+  return self
+end
+
+function _M.rewrite(_, context)
+  local referrer = ngx.var.http_referer
+
+  if referrer then
+    context.extra_params_backend_authrep = context.extra_params_backend_authrep or {}
```

Comment:
We need to find a better way to handle those. They are basically global variables and it starts becoming really ugly.

Reply:
Can't this policy manipulate the proxy object directly?

Reply:
Yes, but the proxy object should also be in `context`. Do you think that's a better solution?

Reply:
I think it already is.

Reply:
Yes. The proxy is stored in `context` in `local_chain.lua` (edited my comment at the same time you were writing).

Now that I think about it, I feel like manipulating here the proxy object is a slightly better solution because it avoids the extra step of doing it in the APIcast policy. Instead of doing that, every policy would be responsible for adding the extra backend params it wants. Is that the benefit you were thinking about?

Reply:
Yes, that would reduce the number of objects we store in the context root. And then changing the params basically becomes proxy public API.

Reply:
Fixed.